src/main/java/org/springframework/hateoas/config/WebTestClientConfigurer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.hateoas.config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.codec.ClientCodecConfigurer;
+import org.springframework.http.codec.json.Jackson2JsonDecoder;
+import org.springframework.http.codec.json.Jackson2JsonEncoder;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.springframework.util.Assert;
+import org.springframework.util.MimeType;
+import org.springframework.web.reactive.function.client.ExchangeStrategies;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Assembles {@link ExchangeStrategies} needed to wire a {@link WebTestClient} with hypermedia support.
+ *
+ * @author Greg Turnquist
+ * @since 1.1
+ */
+@Configuration
+public class WebTestClientConfigurer {
```

Comment:
Framework already has `org.springframework.test.web.reactive.server.WebTestClientConfigurer`. To avoid a clash, I think a HATEOS-specific class name would be good here. Perhaps it could implement Framework's `WebTestClientConfigurer` which would allow it to be passed to `org.springframework.test.web.reactive.server.WebTestClient.Builder.apply(WebTestClientConfigurer)`?

Reply:
Ahh. Thanks Andy. I'll look at that. We often mark stuff up like `HypemediaXXXX`, so perhaps `HypermediaWebTestClientConfigurer` would do.

Let me play around with that extension point and see if I can morph it suitably.